pilot/pkg/config/kube/crdclient/client.go

```diff
@@ -225,9 +228,30 @@ func (cl *Client) UpdateStatus(cfg config.Config) (string, error) {
 		return "", fmt.Errorf("nil status for %v/%v on updateStatus()", cfg.Name, cfg.Namespace)
 	}
 
+	// Special case where IstioStatus isn't properly getting converted to ServiceEntryStatus by the time
+	// UpdateStatus is called by the analyzer. Do the conversion here.
+	//
+	// TODO: Handle ServiceEntryStatus -> IstioStatus analyzer conversion more elegantly before
+	// we get to this point.
```

Comment:
I believe this conversion should happen before calling the geneal `UpdateStatus` method

Reply:
I think John has a better approach here https://github.com/istio/istio/pull/52566. More extensible, the approach here doesn't seem as maintainable especially as new types come into the mix. This was more of a band-aid approach.